src/lib/crypto/krb/string_to_key.c

```diff
@@ -51,8 +51,11 @@ krb5_c_string_to_key_with_params(krb5_context context, krb5_enctype enctype,
         return KRB5_BAD_ENCTYPE;
     keylength = ktp->enc->keylength;
 
+    if (salt == NULL)
```

Comment:
The s2k implementations are inconsistent about whether to allow a null salt.  The DES string-to-key allows it, but the AFS variant does not.  The AES string-to-key allows it, unless pepper is set (as for Camellia and aes-sha2), in which case it does not.  And of course the RC4 string-to-key ignores salt altogether.

make check is failing on this change because lib/kadm5/unit-test/setkey-test.c passes a null salt to krb5_string_to_key() for some DES keys.  I don't think it would be completely safe to error out on null salt in the front end, but it should be safe to substitute a pointer to an empty krb5_data.  That would in turn render all of the null salt conditionals in s2k_des.c and s2k_pbkdf2.c unnecessary.